python/ctsm/site_and_regional/tower_site.py

```diff
@@ -0,0 +1,413 @@
+                    rundir = case.get_value("RUNDIR")
+                    # For existing case check that the compset name is correct
+                    existingcompname = case.get_value("COMPSET")
+                    match = re.search("^HIST", existingcompname, flags=re.IGNORECASE)
+                    # pylint: disable=undefined-variable
+                    if re.search("^HIST", compset, flags=re.IGNORECASE) is None:
+                        expect(
+                            match is None,
+                            """Existing base case is a historical type and should not be
+                            --rerun with the --overwrite option""",
+                        )
+                    # pylint: enable=undefined-variable
+                    else:
+                        expect(
+                            match is not None,
+                            """Existing base case should be a historical type and is not
+                            --rerun with the --overwrite option""",
+                        )
+                    if os.path.isfile(os.path.join(rundir, "ESMF_Profile.summary")):
+                        print("Case {} appears to be complete, not rerunning.".format(case_root))
+                    elif not setup_only:
+                        print("Resubmitting case {}".format(case_root))
+                        case.submit(no_batch=no_batch)
+                        print("-----------------------------------")
+                        print("Successfully submitted case!")
+                        batch_query = self.get_batch_query(case)
+                        if batch_query != "none":
+                            print(f"Use {batch_query} to check its run status")
+                    return
+            else:
+                logger.warning("Case already exists in %s, not overwritting", case_root)
+                return
+        if run_type == "postad":
+            adcase_root = case_root.replace(".postad", ".ad")
+            if not os.path.isdir(adcase_root):
+                logger.warning("postad requested but no ad case found in %s", adcase_root)
+                return
+
+        if not os.path.isdir(case_root):
+            # read_only = False should not be required here
+            with Case(base_case_root, read_only=False) as basecase:
+                print("---- cloning the base case in {}".format(case_root))
+                #
+                # EBK: 11/05/2022 -- Note keeping the user_mods_dirs argument is important. Although
+                # it causes some of the user_nl_* files to have duplicated inputs. It also ensures
+                # that the shell_commands file is copied, as well as taking care of the DATM inputs.
+                # See https://github.com/ESCOMP/CTSM/pull/1872#pullrequestreview-1169407493
+                #
+                basecase.create_clone(case_root, keepexe=True, user_mods_dirs=user_mods_dirs)
+
+        with Case(case_root, read_only=False) as case:
```

Comment:
This section is long and should be broken out into a separate method (could be private). That might make it testable as well.

This would be something done later.

Reply:
#2413